### PR TITLE
Add Initial Security Learning Tutorial

### PR DESCRIPTION
**Pull request by tillmann-crabnebula**
#### What kind of changes does this PR include?

- New or updated content

#### Description

This adds the first security learning tutorial to the `Learn` section.

This was inspired by  https://tauri.by.simon.hyll.nu/labs/chatgpt/1_application/
And the solution component was wildly copied from the source code.

@simonhyll is it fine to use your astro component for `ShowSolution`?
Your project has no explicit license, so I thought I'll rip it and ask for permission and forgiveness before marking this as ready to review 😅 .

### CODE REVIEW

**Comment by simonhyll** [comment]
> @simonhyll is it fine to use your astro component for ShowSolution?

You wouldn't dare! 😲 

Jk ofc x) I've actually already "stolen" it from myself to tauri-docs in my splashscreen lab [here](https://github.com/tauri-apps/tauri-docs/pull/2106/files#diff-0a7f1c6c09899c5645792fa611b7302cd66d83a1d2a0f38ea438848cdd9fd302), so it's already planned to be merged into the repo I just haven't had the time to push the splashscreen lab cross the finish line yet 😅 

I don't really care who gets the honor of committing it to the repo, so the race is on to see who gets their PR merged first and wins the glory of having committed the ShowSolution button, loser has to fix the resulting merge conflict x)

File: src/content/docs/learn/Security/using-plugin-permissions.mdx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+sidebar:
+  badge:
+    text: WIP
+---
+
+import { Steps } from '@astrojs/starlight/components';
+import ShowSolution from '@components/ShowSolution.astro'
+import Cta from '@fragments/cta.mdx';
+
+The goal of this excercise is to get a better understanding on how
```
Comment (dreyfus92) [inline src/content/docs/learn/Security/using-plugin-permissions.mdx]:
```suggestion
The goal of this exercise is to get a better understanding on how
```

File: src/content/docs/learn/Security/writing-plugin-permissions.mdx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+sidebar:
+  badge:
+    text: WIP
+---
+
+import { Steps } from '@astrojs/starlight/components';
+import ShowSolution from '@components/ShowSolution.astro'
+import Cta from '@fragments/cta.mdx';
+
+The goal of this excercise is to get a better understanding on how
```
Comment (dreyfus92) [inline src/content/docs/learn/Security/writing-plugin-permissions.mdx]:
```suggestion
The goal of this exercise is to get a better understanding on how
```

**Review by dreyfus92 — COMMENTED** [review]
Hey @tillmann-crabnebula, I suggested a few minor changes to enhance clarity and consistency. This tutorial is incredibly valuable—thank you for your hard work! 😁

**Comment by tillmann-crabnebula** [comment]
Thanks!
Will do a minor change before asking to merge based on upstream fixes in https://github.com/tauri-apps/tauri/pull/10030/files

**Review by dreyfus92 — APPROVED** [review]
Thanks again @tillmann-crabnebula, LGTM ✅

**Pull request merged**